Source/WebCore/loader/HistoryController.cpp

```diff
@@ -762,11 +761,6 @@ void HistoryController::recursiveUpdateForSameDocumentNavigation()
     if (!m_provisionalItem)
         return;
 
-    // The provisional item may represent a different pending navigation.
-    // Don't commit it if it isn't a same document navigation.
-    if (m_currentItem && !protectedCurrentItem()->shouldDoSameDocumentNavigationTo(*protectedProvisionalItem()))
-        return;
```

Comment:
Should we stop removing this check but move to just before setCurrentItem() by flipping the condition?